### ISSUE TITLE
Env var shorthand notation seems to only support a single env var

### ISSUE THREAD

**Issue by thegedge** [nushell/nushell#1832]
**Describe the bug**
I can only set one environment variable when using the shorthand notation

**To Reproduce**
1. `FOO=bar BAR=test ^env`

**Expected behavior**
Both `FOO` and `BAR` have been set

**Screenshots**

```
❯ FOO=bar BAR=test ^env
error: Command not found
- shell:1:21
1 | FOO=bar BAR=test ^env
  |         ^^^^^^^^ command not found
```

**Configuration (please complete the following information):**
 - OS: macOS 10.15.4
 - Nu: HEAD (5f1136d)
 - Optional features (if any): stable

**Comment by Kelli314** [comment]
I'd like to work on this

**Comment by thegedge** [comment]
There you go, @Kelli314. It's yours 🙂

**Comment by fdncred** [comment]
@Kelli314 Any update on this?

**Comment by Kelli314** [comment]
I've made some progress. The root cause comes from the `with-env` command parsing only ever checking for one variable-value pair [here](https://github.com/nushell/nushell/blob/main/crates/nu-cli/src/commands/with_env.rs#L67). I'm working on how that can be parsed better. Any recommendations would be helpful.

**Comment by jonathandturner** [comment]
@Kelli314 - one possible direction would be to not update `with-env`, and instead wrap with new `with-env` for each variable.

Something like:

```
FOO=1 BAR=2 my command
```

Could expand into:

```
with-env [FOO 1] { with-env [BAR 2] { my command } }
```

**Comment by almindor** [comment]
I just started looking into this and it's a bit of a mess. From what I can tell `process_line` from `nu-cli` already pre-parses the `ENV1=1 ENV2=2` block as `command: ENV1=1` with args: `ENV2=2` which makes this pretty bad to go from.

I'm trying to wrap my head around how this shorthand detection and parsing could work so we don't break things even more.

**Comment by jonathandturner** [comment]
@almindor @Kelli314 - not sure if you ended up collaborating, but I'm happy to answer any questions here or on the discord.

**Comment by jonathandturner** [comment]
Fixed by #3692
